sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/GenerateColumnAccessor.scala

```diff
@@ -114,6 +123,57 @@ object GenerateColumnAccessor extends CodeGenerator[Seq[DataType], ColumnarItera
       (createCode, extract + patch)
     }.unzip
 
+    val accessorCode = accessorClasses.map { case (accessorCls, shortAccCls) =>
+      s"""
+         private $accessorCls get${shortAccCls}(int idx) {
+           byte[] buffer = batch.buffers()[columnIndexes[idx]];
+           return new $accessorCls(ByteBuffer.wrap(buffer).order(nativeOrder));
+         }
+      """
+    }
+    val accessorStructCode = accessorStructClasses.map {
+      case ((accessorCls, dtCls), (shortAccCls, shortDTCls)) =>
+        s"""
+           private $accessorCls get${shortAccCls}_${shortDTCls}(int idx) {
+             byte[] buffer = batch.buffers()[columnIndexes[idx]];
+             return new $accessorCls(ByteBuffer.wrap(buffer).order(nativeOrder),
+               (${dtCls}) columnTypes[idx]);
+           }
+        """
+    }
+
+    /* 4000 = 64000 bytes / 16 (up to 16 bytes per one call)) */
+    val numberOfStatementsThreshold = 4000
```

Comment:
A java method will not be JITted if it's over 8K, so we may need smaller threshold here. Could you also manual check that (for performance)?